controllers/controller/devworkspacerouting/solvers/common_test.go

```diff
@@ -0,0 +1,125 @@
+package solvers
+
+import (
+	"testing"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRouteForEndpointAnnotations(t *testing.T) {
+	tests := []struct {
+		name string
+
+		routingSuffix string
+		endpoint      v1alpha1.Endpoint
+		meta          DevWorkspaceMetadata
+		annotations   map[string]string
+
+		expectedAnnotationsKeys []string
+	}{
+		{
+			name: "nil",
+
+			annotations: nil,
+
+			expectedAnnotationsKeys: []string{
+				"controller.devfile.io/endpoint_name",
+				"haproxy.router.openshift.io/rewrite-target",
+			},
+		},
+		{
+			name: "empty",
+
+			annotations: map[string]string{},
+
+			expectedAnnotationsKeys: []string{
+				"controller.devfile.io/endpoint_name",
+				"haproxy.router.openshift.io/rewrite-target",
+			},
+		},
+		{
+			name: "defined",
+
+			annotations: map[string]string{
+				"example.com/extra": "val",
+			},
+
+			expectedAnnotationsKeys: []string{
+				"controller.devfile.io/endpoint_name",
+				"example.com/extra"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			route := getRouteForEndpoint("routingSuffix", v1alpha1.Endpoint{Name: "Endpoint"}, DevWorkspaceMetadata{DevWorkspaceId: "WorkspaceTest"}, tt.annotations)
+			for _, expected := range tt.expectedAnnotationsKeys {
```

Comment:
Instead of comparing each key (and value) individually, you could also do something like the following to compare the actual vs expected map and get a diff:

`assert.Equal(t, tt.expectedAnnotations, route.Annotations, "Annotations should match: Diff: %s", cmp.Diff( tt.expectedAnnotations, route.Annotations))`